tests/integration/config_test.go

```diff
@@ -209,5 +209,16 @@ var _ = Describe("odo config test", func() {
 			Expect(configValue).To(Not(ContainSubstring(("PORT"))))
 			Expect(configValue).To(Not(ContainSubstring(("SECRET_KEY"))))
 		})
+
+		It("should list config without logging into the OpenShift cluster", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--project", project)
+			helper.CmdShouldPass("odo", "config", "set", "--env", "hello=world")
```

Comment:
I think that even `odo config set` should work without logging to the cluster

Reply:
It doesn't at the moment. Requires login to the OpenShift cluster. Would that mean that config value gets changed in `.odo/config.yml` and requires connection to a cluster only when doing `odo push --config`? Also, should that be a separate issue and PR of its own?

Reply:
yes, to all of it ;-)

Reply:
Can you please open an issue for it, if not already?

Reply:
Done https://github.com/openshift/odo/issues/1791